### DISCUSSION TITLE
Does sms and KGB can support mixed variable?

### DISCUSSION THREAD

**Discussion by muazhari** [anyoptimization/pymoo#621] [Q&A]
![image](https://github.com/user-attachments/assets/c46ec0ba-f808-40ae-98d9-f4dd534055d9)

Specifically, does [kgb](https://github.com/anyoptimization/pymoo/blob/main/pymoo/algorithms/moo/kgb.py) and [sms](https://github.com/anyoptimization/pymoo/blob/main/pymoo/algorithms/moo/sms.py) algorithms support constraints and [mixed variable](https://pymoo.org/customization/mixed.html) by modifying it as in the pict to the below implementation? (yes/no)

Code implementation:
```python
algorithm1 = KGB(
    sampling=MixedVariableSampling(),
    mating=MixedVariableMating(eliminate_duplicates=MixedVariableDuplicateElimination()),
    eliminate_duplicates=MixedVariableDuplicateElimination(),
)

algorithm2 = SMSEMOA(
    sampling=MixedVariableSampling(),
    mating=MixedVariableMating(eliminate_duplicates=MixedVariableDuplicateElimination()),
    eliminate_duplicates=MixedVariableDuplicateElimination(),
)
```

**Comment by blankjul** [comment]
Mixed variables has only been tested as shown in the tutorial. It is not a features that works throughout the whole framework, because the majority of algorthms is only proposed on continouus problem (some of them might apply to any type of problem though)
